server/connectors/slack_connector/client.py

```
@@ -39,6 +43,36 @@ def _get_retry_delay(attempt: int, response=None) -> int:
+            loop = None
+
+        if loop is not None and loop.is_running():
+            # We are inside a running event loop — schedule a coroutine and
+            # block the *current* thread on its result so we don't starve
+            # other coroutines.
+            future = asyncio.run_coroutine_threadsafe(
+                asyncio.sleep(seconds), loop
+            )
+            try:
+                future.result(timeout=seconds + 5)
+            except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)